tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -33,19 +34,24 @@
 
 
 def _GeneratePartitionKey(record: Union[tf.train.Example,
-                                        tf.train.SequenceExample, bytes],
+                                        tf.train.SequenceExample, bytes,
+                                        Dict[str, Any]],
                           split_config: example_gen_pb2.SplitConfig) -> bytes:
   """Generates key for partition."""
 
   if not split_config.HasField('partition_feature_name'):
     if isinstance(record, bytes):
       return record
+    if isinstance(record, dict):
+      return json.dumps(record).encode('utf-8')
     return record.SerializeToString(deterministic=True)
 
   if isinstance(record, tf.train.Example):
     features = record.features.feature  # pytype: disable=attribute-error
   elif isinstance(record, tf.train.SequenceExample):
     features = record.context.feature  # pytype: disable=attribute-error
+  elif isinstance(record, dict):
+    return bytes(record[split_config.partition_feature_name])
```

Comment:
same here, this is not really great

Reply:
it's ok to not support partition_feature_name for now, line 56 will raise error